db/migrate/20190617191109_move_awx_credentials_to_authentications.rb

```diff
@@ -0,0 +1,142 @@
+class MoveAwxCredentialsToAuthentications < ActiveRecord::Migration[5.0]
+  class Authentication < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+    serialize :options
+  end
+
+  class MiqDatabase < ActiveRecord::Base; end
+
+  ENCRYPTED_ATTRIBUTES = %w[
+    auth_key
+    auth_key_password
+    become_password
+    password
+  ].freeze
+
+  FIELD_MAP = {
+    'authorize_password' => 'become_password',
+    'become_username'    => 'become_username',
+    'become_password'    => 'become_password',
+    'password'           => 'password',
+    'secret'             => 'auth_key',
+    'security_token'     => 'auth_key',
+    'ssh_key_data'       => 'auth_key',
+    'ssh_key_unlock'     => 'auth_key_password',
+    'username'           => 'userid',
+    'vault_password'     => 'password'
+  }.freeze
+
+  OPTIONS_FIELDS = %w[
+    authorize
+    become_method
+    client
+    domain
+    host
+    project
+    subscription
+    tenant
+  ].freeze
+
+  def up
+    embedded_ansible_authentications.each do |auth|
+      next unless auth.manager_ref
+
+      say_with_time("Migrating credential #{auth.name} from awx to vmdb") do
+        awx_info = awx_credential_info(auth.manager_ref)
+        update_authentication(auth, awx_info)
+      end
+    end
+  rescue PG::ConnectionBad
+    say("awx database is unreachable, credentials cannot be migrated")
+    return
+  end
+
+  private
+
+  def embedded_ansible_authentications
+    types = %w[
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::Credential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::CloudCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::MachineCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::NetworkCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::ScmCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::VaultCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::AmazonCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::AzureCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::GoogleCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::OpenstackCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::RhvCredential
+      ManageIQ::Providers::EmbeddedAnsible::AutomationManager::VmwareCredential
+    ]
+    Authentication.where(:type => types)
```

Comment:
This should probably be scoped to the current region to avoid the global region trying to migrate credentials from a remote region, right?